src/data/polynomial/eval.lean

```diff
@@ -779,6 +784,10 @@ begin
   exact ring_hom.map_zero f,
 end
 
+lemma is_root_map {R : Type*} [comm_ring R] {f : R →+* S} {x : R} {p : polynomial R}
+  (hf : function.injective f) : is_root p x ↔ is_root (p.map f) (f x) :=
```

Comment:
Please swap the two sides, and add `iff` to the name.
I think `is_root.map` should be the lemma that `f x` is a root of `p.map f` (which doesn't need `hf`). Would you mind adding that lemma as well?

Reply:
I've done this, and also added the converse - I think there is an asymmetry required in the converse, however, so just want to check that that's required or maybe there's a slicker proof I'm missing.